app/main.py

```diff
@@ -1,3 +1,14 @@
 def cache(func):
-    # Write your code here
-    pass
+    functions = dict()
+    arguments = set()
+
+    def inner(*args):
+        if func.__name__ in functions and args in arguments:
+            print("Getting from cache")
+            return functions[func.__name__]
+        else:
+            print("Calculating new result")
+            functions[func.__name__] = func(*args)
```

Comment:
Dictionary should store results for different **args** not for different **functions**